frontend/src/components/Table/ColumnFilter.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import { ColumnInstanceWithProps } from '.';
+import styled from 'styled-components';
+import { Form } from 'react-bootstrap';
+import ClickAwayListener from 'react-click-away-listener';
+import { getIn, useFormikContext } from 'formik';
+import clsx from 'classnames';
+import { FiFilter } from 'react-icons/fi';
+import { FaFilter } from 'react-icons/fa';
+
+interface IColumnFilterProps {
+  column: ColumnInstanceWithProps<any>;
+  onFilter: (values: any) => void;
+}
+
+const Wrapper = styled('div')`
+  display: flex;
+  align-items: center;
+  justify-items: center;
+  position: relative;
+  font-size: 12px;
+`;
+
+const InputContainer = styled('div')`
+  position: absolute;
+  bottom: -55px;
+  left: 0;
+  z-index: 1;
+
+  .form-control {
+    border-radius: 0;
+    border-color: #606060;
+  }
+`;
+
+const ColumnFilter: React.FC<IColumnFilterProps> = ({ column, onFilter, children }) => {
+  const [open, setOpen] = React.useState(false);
+  const context = useFormikContext();
+  const handleClose = React.useCallback(() => {
+    onFilter(context.values);
+    setOpen(false);
+  }, [onFilter, context.values, setOpen]);
+
+  if (!column.filterable) {
+    return <>{children}</>;
+  }
+
+  const hasValue = !!getIn(context.values, column.filter?.props!.name);
+  const Control = (column.filter?.component || Form.Control) as any;
+
+  return (
+    <Wrapper
+      className={clsx('filter-wrapper', { active: hasValue })}
+      onClick={e => {
+        if ((e.target as any)?.className?.indexOf('filter-wrapper') !== -1) {
```

Comment:
won't this onClick event always have a target element with classname of filter-wrapper? the clsx statement above should create two classnames, 'filter-wrapper' and then conditionally 'active' if hasValue is true.